### ISSUE TITLE
Fair Regression Metrics

### ISSUE THREAD

**Issue by albertxi1** [hil-se/FairReweighing#1]
**_Fair Regression: Quantitative Definitions and Reduction-based Algorithms_**

Statistical parity: f(X) is independent of the protected attribute A.
<img width="529" alt="image" src="https://user-images.githubusercontent.com/45436233/232624908-e2a9bdd4-73b6-438f-8053-d21463b26b32.png">

- Discrete SA
- Did not consider accuracy
- Ludicrous when there is a causal relationship between the protected attribute and the output variable

Bounded group loss (BGL):
<img width="329" alt="image" src="https://user-images.githubusercontent.com/45436233/232625471-f44a81e4-c04d-429c-81bc-4cb236b15b08.png">

- Discrete SA
- Have to pick loss function

**Comment by albertxi1** [comment]
A Convex Framework for Fair Regression (Richard et al., 2017)

- Individual Fairness
<img width="319" alt="Screen Shot 2021-11-15 at 17 43 21" src="https://user-images.githubusercontent.com/45436233/141864366-7cc3eb20-1971-45ec-b787-d59bd85ca318.png">

- Group Fairness
<img width="332" alt="Screen Shot 2021-11-15 at 17 44 10" src="https://user-images.githubusercontent.com/45436233/141864449-f0802afd-6a8b-4d7b-b4b8-7488ddc33193.png">

- Hybrid notions of fairness
<img width="524" alt="Screen Shot 2021-11-15 at 17 44 32" src="https://user-images.githubusercontent.com/45436233/141864490-84459447-362f-4b8b-afc2-0dda83e71cc2.png">
- <img width="458" alt="Screen Shot 2021-11-15 at 18 00 53" src="https://user-images.githubusercontent.com/45436233/141866224-528186b9-02a9-49bf-acaa-91128799c9e2.png">

- Minimizing<img width="185" alt="Screen Shot 2021-11-15 at 17 46 44" src="https://user-images.githubusercontent.com/45436233/141864714-71c0867c-0ef5-4da0-81c4-ead17b186c44.png">, in which f is the fairness penalty, and use l2 regularization to prevent overfitting.

- Comparing pairs of instances (one from each group) as cross pairs, essential Individual fairness, all under the notion that similar instances should be treated similarly

- Difference is that whether unfairness can be cancelled or not
- D(yi,yj) is inversely proportional to | yi − yj |
- Does not rule out perfect predictor (Like AOD and EOD)
- Whether to use sensitive attributes in learning process does not affect much
- Transfer classification to regression by Pr[yi = 1] = exp(w · xi)/(1 + exp(w · xi))

https://arxiv.org/abs/1706.02409

<img width="699" alt="Screen Shot 2022-02-09 at 9 08 46 AM" src="https://user-images.githubusercontent.com/13929197/153217729-c24d5e0b-c3f1-4126-bbad-4ae878d27fb7.png">
